packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -132,79 +136,118 @@ export default class Proxy {
     return xapi.rebootVm(vmUuid)
   }
 
-  async deployProxy(srId, { networkConfiguration, networkId, proxyId } = {}) {
+  @defer
+  async _createProxyVm(
+    $defer,
+    srId,
+    licenseId,
+    { networkId, networkConfiguration }
+  ) {
     const app = this._app
-
-    const redeploy = proxyId !== undefined
-    if (redeploy) {
-      const { vmUuid } = await this._getProxy(proxyId)
-      if (vmUuid !== undefined) {
-        await app.getXapi(vmUuid).deleteVm(vmUuid)
-        await this.updateProxy(proxyId, {
-          vmUuid: null,
-        })
-      }
-    }
-
     const xoProxyConf = this._xoProxyConf
 
     const namespace = xoProxyConf.namespace
     const {
       [namespace]: { xva },
     } = await app.getResourceCatalog()
     const xapi = app.getXapi(srId)
-    let vm = await xapi.importVm(
+    const vm = await xapi.importVm(
       await app.requestResource({
         id: xva.id,
         namespace,
         version: xva.version,
       }),
       { srId }
     )
-    let date, proxyAuthenticationToken, xenstoreData
-    try {
-      if (networkId !== undefined) {
-        await Promise.all([
-          ...vm.VIFs.map(vif => xapi.deleteVif(vif)),
-          xapi.createVif(vm.$id, networkId),
-        ])
-      }
+    $defer.onFailure(() => xapi._deleteVm(vm))
 
-      date = new Date()
-      proxyAuthenticationToken = await generateToken()
-
-      const [
-        password,
-        { registrationToken, registrationEmail: email },
-      ] = await Promise.all([generateToken(10), app.getApplianceRegistration()])
-      xenstoreData = {
-        'vm-data/system-account-xoa-password': password,
-        'vm-data/xo-proxy-authenticationToken': JSON.stringify(
-          proxyAuthenticationToken
-        ),
-        'vm-data/xoa-updater-credentials': JSON.stringify({
-          email,
-          registrationToken,
-        }),
-        'vm-data/xoa-updater-channel': JSON.stringify(xoProxyConf.channel),
-      }
-      if (networkConfiguration !== undefined) {
-        xenstoreData['vm-data/ip'] = networkConfiguration.ip
-        xenstoreData['vm-data/gateway'] = networkConfiguration.gateway
-        xenstoreData['vm-data/netmask'] = networkConfiguration.netmask
-        xenstoreData['vm-data/dns'] = networkConfiguration.dns
-      }
+    const props = { licenseId, boundObjectId: vm.uuid }
```

Comment:
`props`?

Reply:
`props` is often used in the React context, AFAICT, it's rarer in the JS word, maybe `opts` or `args`?

Reply:
After doing a simple search, i saw that it used in `xen-api`, `backup-ng`, ... folders. So it is a rule that we should be based on it in the future ?  Never use `props` in the server side?

Reply:
In these cases, they are used in the context of object properties :slightly_smiling_face: